zetaclient/bitcoin/bitcoin_client.go

```diff
@@ -167,9 +166,12 @@ func NewBitcoinClient(
 	ob.includedTxHashes = make(map[string]bool)
 	ob.includedTxResults = make(map[string]*btcjson.GetTransactionResult)
 	ob.broadcastedTx = make(map[string]string)
-	ob.params = btcCfg.ChainParams
-
+	_, chainParams, found := appcontext.ZetaCoreContext().GetBTCChainParams()
+	if found {
+		ob.params = *chainParams
```

Comment:
I think we should not create this instance of `BTCChainClient` if `found == false`. And that's why previous constructor explicitly takes a `config.BTCConfig` argument. For future unit tests (we don't have yet), passing a `config.BTCConfig` argument in constructor probably makes us easier to create a dummy/mock BTCChainClient object.